sdk/java-sdk/src/main/scala/kalix/javasdk/impl/RestDeferredCall.scala

```diff
@@ -31,9 +33,19 @@ final case class RestDeferredCall[I, O](
     methodName: String,
     asyncCall: () => CompletionStage[O])
     extends DeferredCall[I, O] {
-  override def execute(): CompletionStage[O] = asyncCall()
+  override def execute(): CompletionStage[O] = asyncCall().exceptionally {
+    case responseException: WebClientResponseException =>
+      throw DeferredCallResponseException(
+        responseException.getMessage,
+        StatusCodeConverter.fromWebClientResponse(responseException),
+        responseException)
+    case other: Throwable => throw other
```

Comment:
That would also work for gRPC calls, no? 

I mean, if I'm executing a `GrpcDeferredCall` we could also convert it to a Kalix ErrorCode so the types that I'm exposed to will be the same as when I call `effect.error()`. 

The only problem is that such a change is backward incompatible. Code that is already handling this kind of exceptions is already expecting `io.grpc.Status`. So we can't change it. 

With the current solution, code-first SDK using http, will be exposed to `ErrorCode` on both ends, but Protobuf SDK will have `ErrorCode` when defining errors, but `io.grpc.Status` when handling errors.

Reply:
Yeah, that is also one reason I would argue for not deprecating error() with Status.Code. I think I would just keep both APIs.